newIDE/app/src/VariablesList/VariablesList.js

```diff
@@ -0,0 +1,1360 @@
+          if (newBool === _variable.getBool()) return;
+          break;
+        default:
+      }
+      const newVariable = new gd.Variable();
+      unserializeFromJSObject(newVariable, serializeToJSObject(_variable));
+      variable = props.variablesContainer.insert(name, newVariable, 0);
+      const newSelectedNodes = [...selectedNodes];
+      const isVariableSelected = newSelectedNodes.indexOf(nodeId) !== -1;
+      if (isVariableSelected) {
+        newSelectedNodes.splice(newSelectedNodes.indexOf(nodeId), 1, name);
+        setSelectedNodes(newSelectedNodes);
+      } else {
+        setSelectedNodes([...newSelectedNodes, name]);
+      }
+      newVariable.delete();
+    } else {
+      const { variable: _variable } = getVariableContextFromNodeId(
+        nodeId,
+        props.variablesContainer
+      );
+      variable = _variable;
+    }
+    if (!variable) return;
+    switch (variable.getType()) {
+      case gd.Variable.String:
+        if (variable.getString() === newValue) return;
+        variable.setString(newValue);
+        break;
+      case gd.Variable.Number:
+        const newValueAsFloat = parseFloat(newValue);
+        if (newValueAsFloat === variable.getValue()) return;
+        variable.setValue(newValueAsFloat);
+        break;
+      case gd.Variable.Boolean:
+        const newBool = newValue === 'true';
+        if (newBool === variable.getBool()) return;
+        variable.setBool(newBool);
+        break;
+      default:
+        console.error(
+          `Cannot set variable with type ${variable.getType()} - are you sure it's a primitive type?`
+        );
+    }
+    _saveToHistory();
+    forceUpdate();
+  };
+
+  const renderTree = (isInherited: boolean = false) => {
+    const variablesContainer =
+      isInherited && props.inheritedVariablesContainer
+        ? props.inheritedVariablesContainer
+        : props.variablesContainer;
+    const containerVariablesTree = mapFor(
+      0,
+      variablesContainer.count(),
+      index => {
+        const variable = variablesContainer.getAt(index);
+        const name = variablesContainer.getNameAt(index);
+        if (isInherited) {
+          if (props.variablesContainer.has(name)) {
+            return null;
+          }
+        }
+
+        return renderVariableAndChildrenRows({
+          name,
+          variable,
+          isInherited,
+        });
+      }
+    );
+    return containerVariablesTree;
+  };
+
+  return (
+    <ClickAwayListener onClickAway={() => setSelectedNodes([])}>
+      <Measure
+        bounds
+        onResize={contentRect => {
+          setContainerWidth(contentRect.bounds.width);
+        }}
+      >
+        {({ contentRect, measureRef }) => (
+          <div
+            ref={measureRef}
+            style={{ flex: 1, display: 'flex' }}
+            onKeyDown={keyboardShortcuts.onKeyDown}
+            onKeyUp={keyboardShortcuts.onKeyUp}
+          >
+            <Column expand reverse={isNarrow}>
```

Comment:
Instead of reverse, could you put `<VariablesListToolbar ...>` in a variable and put it at the right place?

I'm a bit worried of increasing the API surface of Column.